tsconfig.json

```diff
@@ -1,12 +1,16 @@
 {
-	"compilerOptions": {
-		"module": "commonjs",
-		"target": "ES5",
-		"outDir": "out",
-		"noLib": true,
-		"sourceMap": true
-	},
-	"exclude": [
-		"node_modules"
-	]
+  "compilerOptions": {
+    "baseUrl": ".",
+    "esModuleInterop": true,
+    "lib": ["es5", "es2015", "es2017"],
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "noImplicitAny": true,
+    "noImplicitThis": false,

```

Comment:
what happens if we enable it?

Reply:
it will raise error on `this` expressions with an implied `any` type. it is just a convenience. i would remove this unnecessary this codebase